### PR TITLE
Remove mbed wait deprecated APIs

### PR DESCRIPTION
**Pull request by rajkan01**
### Summary of changes 
Removed mbed wait deprecated APIs.

#### Impact of changes 
Breaking change: `wait` and `wait_ms` APIs has been deprecated since Mbed OS 5.14 and they are removed now.

#### Migration actions required 
Use ` ThisThread::sleep_for`, `thread_sleep_for`, and `wait_us` APIs.

### Documentation 
None

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [x] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 

    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers 
@evedon

----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: platform/mbed_wait_api.h [hunk 1]
```diff
@@ -46,54 +46,14 @@ extern "C" {
  * int main() {
  *     while (1) {
  *         heartbeat = 1;
- *         wait(0.5);
+ *         wait_us(5000);
```
Comment (evedon) [inline platform/mbed_wait_api.h]: Why not `thread_sleep_for`?

Comment (rajkan01) [reply platform/mbed_wait_api.h]: This example code is to show usage of the wait API, so I have used wait_us API.

Comment (evedon) [reply platform/mbed_wait_api.h]: Not this example.
If you want to demonstrate usage of `wait_us` then you need an example of wait from ISR context or example of spin waiting.

Comment (evedon) [reply platform/mbed_wait_api.h]: or perhaps wait for less than 1ms

File: TESTS/mbed_hal_fpga_ci_test_shield/pwm/main.cpp [hunk 2]
```diff
@@ -127,11 +127,11 @@ void fpga_pwm_period_fill_test(PinName pin, uint32_t period_ms, uint32_t fill_pr
             break;
     }
 
-    ThisThread::sleep_for(period_ms);
+    wait_us(PERIOD_US(period_ms));
```
Comment (evedon) [inline TESTS/mbed_hal_fpga_ci_test_shield/pwm/main.cpp:131]: Please add a comment to say why wait_us is required here

**Pull request closed without merging**